cli/lib/kontena/cli/stacks/yaml/stack_file_loader/file_loader.rb

```diff
@@ -0,0 +1,40 @@
+require 'pathname'
+
+module Kontena::Cli::Stacks
+  module YAML
+    class FileLoader < StackFileLoader
+      def self.match?(source, parent = nil)
+        ::File.exist?(with_context(source, parent))
+      end
+
+      def self.is_file?(parent)
+        parent.is_a?(FileLoader)
+      end
+
+      def self.with_context(source, parent = nil)
+        if is_file?(parent)
```

Comment:
Wut, is this like pulling in a stack from the registry, and then that `depends: stack: ~/.ssh/id_rsa`? Should this perhaps skip the match / fail with an exception if trying to depend on a local file stack from a remote stack?

Reply:
This trickery is to figure out parent directory so that `foo/xyz.yml` is searched from `parent_dir/foo/xyz.yml`.

The registry commands refuse to push with local dependencies or extends.  Of course you can push stacks like that using some other client.

(Trying to parse `id_rsa` as YAML returns a string and the parser will crash because it was expecting a hash.)